lib/membrane_http_adaptive_stream/source/client_genserver.ex

```diff
@@ -104,41 +122,34 @@ defmodule Membrane.HTTPAdaptiveStream.Source.ClientGenServer do
   end
 
   @impl true
-  def handle_cast({:request_audio_chunk, pid}, state) do
-    {chunk, client} = Client.read_audio_chunk(state.client)
-
-    state =
-      put_in(
-        state.how_much_truly_skipped,
-        Client.get_how_much_truly_skipped_ms(client) |> Membrane.Time.milliseconds()
-      )
+  def handle_cast({:request_chunk_or_eos, pid}, state) do
+    # this is ugly but this is how StreamSplit works
+    {chunk_or_eos, stream} =
+      try do
+        state.stream |> StreamSplit.pop()
+      rescue
+        _e in MatchError -> {:end_of_stream, state.stream}
+      end
```

Comment:
Couldn't you check if the stream is empty with `Enum.empty?` and call `StreamSplit.pop()` only if it's not?

Reply:
I guess Enum.empty will try to pop an element or iterate over it or sth like this